xbmc/cores/RetroPlayer/rendering/VideoShaders/IVideoShaderPreset.h

```diff
@@ -113,15 +113,58 @@ namespace SHADER
     virtual ~IVideoShaderPreset() = default;
 
     //todo: impl once and for all
+   /*!
+    * \brief Reads/Parses a shader preset file and loads its state to the
+    *        object. What this state is is implementation specific.
+    * \param presetPath Full path of the preset file.
+    * \return True on successful parsing, false on failed.
+    */
     virtual bool ReadPresetFile(const std::string &presetPath) = 0;
 
+    /*!
+     * \brief Updates state if needed and renderes the preset to the target texture
+     * \param dest Coordinates of the 4 corners of the output viewport/window
+     * \param source Input texture. The source of the video frame, in is original resolution (unscaled)
+     * \param target The target texture. The texture that the final result will be rendered in.
+     * \return Returns false if updating or rendering failed, true if both succeeded.
+     */
     virtual bool RenderUpdate(CPoint dest[], IShaderTexture* source, IShaderTexture* target) = 0;
+
+    /*!
+     * \brief Informs about the speed of playback
+     * \param speed Commonly 1.0 for normal playback, and 0.0 if paused
+     */
     virtual void SetSpeed(double speed) = 0;
-    virtual void SetVideoSize(const unsigned videoWidth, const unsigned videoHeight) = 0;
+
+    /*!
+     * \brief Size of the input/source frame in pixels
+     * \param videoWidth Height of the source frame in pixels
+     * \param videoHeight Height of the source frame in pixels
+     */
+    virtual void SetVideoSize(const unsigned videoWidth, const unsigned videoHeight);
+
+    /*!
+     * \brief Set the preset to be rendered on the next frame
+     * \param shaderPresetPath Full path to the preset file to be loaded
+     * \return Returns false if loading the preset failed, true otherwise.
+     */
     virtual bool SetShaderPreset(const std::string& shaderPresetPath) = 0;
+
+    /*!
+     * \brief Gets the full path to the shader preset
+     * \return The full path to the currently loaded preset file
+     */
     virtual const std::string& GetShaderPreset() const = 0;
 
+    /*!
+     * \brief Gets the passes of the loaded preset
+     * \return All the video shader passes of the currently loaded preset
+     */
     virtual VideoShaderPasses& GetPasses() = 0;
   };
+
+  inline void IVideoShaderPreset::SetVideoSize(const unsigned videoWidth, const unsigned videoHeight)
```

Comment:
Why is this here?

Reply:
Whoops, leftovers from ReSharper, I'll take care of it later today.